### PR TITLE
StatsCollector, Aggregators, MetricsConfiguration move

### PR DESCRIPTION
**Pull request by Tjofil**
#2 

This PR focuses on extracting three major components/component groups:

- `MetricsConfiguration` is a wrapper class around map that serves as configuration registry for all the collectors, both ones stored in `PA` and ones stored in `PA-RCA` repo. We don't do any initialization of the map entries from commons as that would require the class to know about collectors, which introduces dependencies from the two repos. Instead, we already have [an initialization of this map from PA side](https://github.com/opensearch-project/performance-analyzer/blob/f65f8f91928c39ada3665079656ac59e82cda704/src/main/java/org/opensearch/performanceanalyzer/util/Utils.java#L29) which is done upon plugin class load, and [the current initialization from inside class](https://github.com/opensearch-project/performance-analyzer-rca/blob/a6edfbf6d3d71511e89c8c2d418f1582cbeb5407/src/main/java/org/opensearch/performanceanalyzer/metrics/MetricsConfiguration.java#L47) is going to be replaced with a symmetrical one that happens on `PerformanceAnalyzerApp` class load.

- `SampleAggregators` used to live as `PerformanceAnalyzerApp`'s static fields and Writer has to reach for this class in order to write to aggregators. We extract them in this PR, alongside `StatsReporter` by whom they are consumed, to a class called `CommonStats` which is visible to both `PA` and `PA-RCA` repo and thus eliminating `PA`'s need to know about the PA app class. They're still going to be initialized from `PA-RCA` code but used by both repos.

- `StatsCollector` and everything necessary for it's functioning has also been moved to the repo, no changes were done to these classes. Note that `PA` and `PA-RCA` have their own tests for different use cases of `StatsCollector` and these tests will remain inside their respective repos.

Besides these extractions there is also a fix for Lychee link checker and tests for classes that had them inside their original repos.

There's also been a change to packages, generally small moves and flattening of the hierarchy and removal of `rca` subpackage as it was misleading in most of the cases.

After this I'll publish two branches, one for PA and second for PA-RCA and make the changes to account for moving of the `MetricsConfiguration`, aggregators and StatsCollector so we can incrementally change the repos and make sure everything keeps working from their point of view.

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/performanceanalyzer/commons/metrics/MetricsConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            this.samplingInterval = samplingInterval;
+            this.rotationInterval = rotationInterval;
+        }
+    }
+
+    public static final Map<Class, MetricConfig> CONFIG_MAP = new HashMap<>();
+    public static final MetricConfig cdefault;
+
+    static {
+        cdefault = new MetricConfig(SAMPLING_INTERVAL, 0);
```
Comment (khushbr) [inline src/main/java/org/opensearch/performanceanalyzer/commons/metrics/MetricsConfiguration.java:31]: Let us add a small comment here repeating what is mentioned in PR comment -  With this refactoring, the CONFIG_MAP has been moved out and will be statically initialized within the `PerformanceAnalyzerApp`.

File: src/main/java/org/opensearch/performanceanalyzer/commons/stats/CommonStats.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.commons.stats;
+
+/* Catalogue class that is to be populated upon PerformanceAnalyzerApp class load
+and to be used by both PA and PA-RCA as well as commons repo.
```
Comment (khushbr) [inline src/main/java/org/opensearch/performanceanalyzer/commons/stats/CommonStats.java:9]: Can we update the comment as : 

" Catalog *Operational Metrics* class that is to be populated upon PerformanceAnalyzerApp class load and to be used by both PA and PA-RCA. "

One other question, is this class used at runtime by the commons repo ?

**Comment by khushbr** [comment]
Thank you for making the changes @Tjofil. One high level comment I have is around naming of the sub-directories:

Would it make sense to:
1. Rename `collectors` -> `statcollector` as it logically is a single piece for collecting operations stats.
2. `format` -> `formatter`
3. Move `src/main/java/org/opensearch/performanceanalyzer/commons/stats/impl/vals/AggregateValue.java` to 1 level up  -> `src/main/java/org/opensearch/performanceanalyzer/commons/stats/vals/AggregateValue.java`

File: src/main/java/org/opensearch/performanceanalyzer/commons/collectors/StatExceptionCode.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    READER_RESTART_PROCESSING("ReaderRestartProcessing"),
+    RCA_SCHEDULER_RESTART_PROCESSING("RCASchedulerRestartProcessing"),
+    RCA_NETWORK_ERROR("RcaNetworkError"),
+    RCA_VERTEX_RX_BUFFER_FULL_ERROR("RcaVertexRxBufferFullError"),
+    RCA_NETWORK_THREADPOOL_QUEUE_FULL_ERROR("RcaNetworkThreadpoolQueueFullError"),
+    RCA_SCHEDULER_STOPPED_ERROR("RcaSchedulerStoppedError"),
+    OPENSEARCH_REQUEST_INTERCEPTOR_ERROR("OpenSearchRequestInterceptorError"),
+    MISCONFIGURED_OLD_GEN_RCA_HEAP_MAX_MISSING("MisconfiguredOldGenRcaHeapMaxMissing"),
+    MISCONFIGURED_OLD_GEN_RCA_HEAP_USED_MISSING("MisconfiguredOldGenRcaHeapUsedMissing"),
+    MISCONFIGURED_OLD_GEN_RCA_GC_EVENTS_MISSING("MisconfiguredOldGenRcaGcEventsMissing"),
```
[the remark this answers is not in the document]
Comment (Tjofil) [reply src/main/java/org/opensearch/performanceanalyzer/commons/collectors/StatExceptionCode.java:47]: Could you be more specific ?

**Comment by Tjofil** [comment]
@khushbr 
With 2nd and 3rd I completely agree.
Besides `StatsCollector` we also have `PerformanceAnalyzerMetricsCollector` and `Version` classes inside collectors that are to be used by all collectors, we can only move the `StatsCollector` out if that's what we want but to me this current setup seems fine as other collectors will also live in the collectors package.

**Comment by khushbr** [comment]
> @khushbr With 2nd and 3rd I completely agree. Besides `StatsCollector` we also have `PerformanceAnalyzerMetricsCollector` and `Version` classes inside collectors that are to be used by all collectors, we can only move the `StatsCollector` out if that's what we want but to me this current setup seems fine as other collectors will also live in the collectors package.

This sounds good to me, let us retain the name of repo as `collectors`

**Review by khushbr — APPROVED** [review]
LGTM! Ship it!

**Review by ansjcy — APPROVED** [review]

**Pull request merged**